### PR TITLE
Modify module files to compile on AWS with spack-stack-1.6.0

### PR DESCRIPTION
**Pull request by weihuang-jedi**
Modify module files to compile on AWS with spack-stack-1.6.0

Solve issue: #2257 2257
(https://github.com/ufs-community/ufs-weather-model/issues/2257)

This code change is in the effort to implement global-workflow on AWS and other NOAA Cloud machines, and use spack-stack 1.6.0.
  
---

## Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] Jet
  - [ ] Gaea
  - [ ] Derecho
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by jkbk2004** [comment]
@weihuang-jedi this pr can be combined to #2093

File: CMakeLists.txt [hunk 1]
```diff
@@ -144,8 +144,10 @@ if(OPENMP)
   find_package(OpenMP REQUIRED)
 endif()
 
-find_package(NetCDF 4.7.4 REQUIRED C Fortran)
-find_package(ESMF 8.3.0 MODULE REQUIRED)
+#find_package(NetCDF 4.7.4 REQUIRED C Fortran)
+#find_package(ESMF 8.3.0 MODULE REQUIRED)
```
Comment (RatkoVasic-NOAA) [inline CMakeLists.txt]: Please remove old lines with comments.

Comment (DusanJovic-NOAA) [reply CMakeLists.txt]: Why are version numbers removed?

**Comment by weihuang-jedi** [comment]
Old lines with comments removed.

**Comment by weihuang-jedi** [comment]
Right now, current UFS-model use spack-stack 1.5.1, AWS uses 1.6.0, which
use different NetCDF and ESMF libs. Remove versions will relax those
version requirements, but just the library (if any version exist). So,
unless UFS-model depends on specific version of those libs, version is not
necessary. Also we load those libs through module files, where we define
the version numbers.

Or, as there is a PR 2093, is update spack-stack libs to 1.6.0, we can wait
after that PR is merged, and put back the versions.

**Comment by RatkoVasic-NOAA** [comment]
@weihuang-jedi as @jkbk2004 said, how about you create PR to #2093? In that case you can replace old ufs_common.lua with your new one (so we have one less file). We will run tests again to see how ufs works on other machines with new ESMF, FMS, NETCDF, SP and MAPL. And we can keep app version numbers in CMakeLists.txt file.
That would be PR to this branch: https://github.com/RatkoVasic-NOAA/ufs-weather-model/tree/ss-160

**Comment by weihuang-jedi** [comment]
Actually, I think combine together is not straight forward, as my code is
branched from fork:
https://github.com/NOAA-EPIC/ufs-weather-model-cloud/tree/wei-epic-aws
Easy way is still #2093 merge first, then I can sync my code, and ask for
merge again.
Other wise, we probably need to manually copy over code, or I do not how to
merge two forked code together.
After #2093 merged in, I'll try to not modify CMakeLists.txt, and remove
ufs_common4noaaloud.lua but keep other changes.

Will this be OK?

**Comment by RatkoVasic-NOAA** [comment]
@weihuang-jedi, that works with me as well.
@jkbk2004 are there any news on WCOSS2 installing spack-stack-1.6.0? My branch ss-170 is ready for PR for spack-stack-1.7.0 already.

**Pull request closed without merging**